### DISCUSSION TITLE
Turn JSBSim into a 3DoF simulator

### DISCUSSION THREAD

**Discussion by radoof** [JSBSim-Team/jsbsim#505] [Q&A]
Hello everyone! 

I am looking into doing some trajectory simulations where I would like to not have any rotational motion (so basically I want a 3DoF simulator). Is there any way to achieve that?

Thank you!

**Comment by seanmcleod** [comment]
You could create an aircraft model with all forces at the cg, in which case it'll move like a UFO.

**Reply by radoof** [reply]
Thank you for the reply @seanmcleod ! I have a follow-up question related to this. I now have the plane model defined with all forces acting on CG, and no moment coefficients. I have an AoA and flight path angle profile that I would like to run. I've tried directly setting the angle of attack and flight path angle in JSBSim, but that didn't work (I want to vary them myself during the simulation, so I'm not looking at just initial conditions). Is there a nice way to do this? (I realize JSBSim is not meant for this, but bare with me 😅 )

Thank you!

**Comment by seanmcleod** [comment]
Hmm, so typically you would adjust AoA and flight path using some combination of thrust and adjusting pitch attitude via a pitch moment. But you've got no moments so you can't change the initial pitch attitude you start off with, so you're only left with thrust as a control input?

**Reply by radoof** [reply]
Yup, that would be the case. So now I would have the engine thrust that is pointing in the direction of the initial pitch attitude, without any way of changing the thrust direction (could do something like adding another thruster that is orthogonal and change the thrust vector direction like that, but then I'm not changing the attitude of the plane). 

The plane is not powered throughout the entire flight, so in this case I would be using the AoA profile for the gliding flight, and since I don't have any moments to change the attitude of the plane, I would want to directly change the AoA so that I maintain some CL/CD ratio. Thoughts?

**Reply by bcoconni** [reply]
> I would want to directly change the AoA so that I maintain some CL/CD ratio. Thoughts?

If your goal is just to modify the AoA then another option is to change the relative wind direction. This can be achieved by altering one or several of the following properties:
* `atmosphere/wind-north-fps`
* `atmosphere/wind-east-fps`
* `atmosphere/wind-down-fps`

**Comment by seanmcleod** [comment]
I'm wondering whether JBSim is the best option. Why not write your own code to simply sum the forces, with the ability in your code to set/change the AoA at any time step, then integrate them to get velocity and integrate again to get your trajectory position. Maybe you borrow the atmosphere model if you want to take into account varying atmospheric conditions based on altitude etc.

You're trying to generate/model trajectories of a glider? Do you need to model thermals?

**Comment by radoof** [comment]
@seanmcleod @bcoconni Thank you for your replies and help! Going to go with a custom simulator for this particular application. 😃
